.github/workflows/ci.yml

```diff
@@ -12,6 +12,9 @@ on:
 concurrency:
   group: ${{ github.workflow }}-${{ github.actor }}-${{ github.head_ref || github.run_id }}
   cancel-in-progress: true
+permissions:
+  contents: read  # for actions/checkout to fetch code  
+  id-token: write # for requesting the JWT
```

Comment:
refer to https://github.com/karmada-io/karmada/pull/5119#discussion_r1662065591, the write permissions can be declared at the [run-level](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idpermissions). Also, can you confirm whether write permission is required? The result given by the [StepSecurity's online tool](https://app.stepsecurity.io/secureworkflow/) is that write permission is not required.

Reply:
That would be correct, write permission is not necessary